be/src/formats/parquet/column_reader.cpp

```diff
@@ -348,7 +352,22 @@ class StructColumnReader : public ColumnReader {
         return Status::InternalError("No existed parquet subfield column reader in StructColumn");
     }
 
-    Status prepare_batch(size_t* num_records, ColumnContentType content_type, Column* dst) override {
+    StatusOr<size_t> skip(const size_t rows_to_skip) override {
+        size_t rows_skip = 0;
+        for (const auto& reader : _child_readers) {
+            if (reader != nullptr) {
+                if (rows_skip == 0) {
+                    ASSIGN_OR_RETURN(rows_skip, reader->skip(rows_to_skip));
```

Comment:
why special check here?

Reply:
Child reader may be nullptr for ComplexTypes, and that only happens when doing Iceberg Schema Evolution. Like iceberg add a new subfield in a struct column, that newly added subfield column reader will be nullptr when reading the previous parquet file(because that subfield has not existed in the previous parquet file), so we will `append_default()` for a non-existed subfield column.

You can see this logic in `prepare_batch()`.